jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +303,30 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    @with_spinner()
```

Comment:
I don't think this is used anywhere and `collapse_to_spectrum` (which we should probably consider renaming) already handles the spinner.

```suggestion
```